desktop_app/src/ui/components/Chat/SlashCommandDropdown/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { Command, Hash, Zap } from 'lucide-react';
+import React from 'react';
+
+import { SlashCommand } from '@ui/lib/utils/slash-commands';
+import { cn } from '@ui/lib/utils/tailwind';
+
+interface SlashCommandSuggestion {
+  command: SlashCommand;
+  description: string;
+}
+
+interface SlashCommandDropdownProps {
+  suggestions: SlashCommandSuggestion[];
+  selectedIndex: number;
+  onSelect: (command: SlashCommand) => void;
+  visible: boolean;
+  inputRect?: DOMRect;
+}
+
+const getCommandIcon = (command: SlashCommand) => {
+  switch (command) {
+    case SlashCommand.CLEAR:
+      return <Hash className="w-4 h-4" />;
+    case SlashCommand.COMPACT:
+      return <Zap className="w-4 h-4" />;
+    default:
+      return <Command className="w-4 h-4" />;
+  }
+};
+
+export function SlashCommandDropdown({
+  suggestions,
+  selectedIndex,
+  onSelect,
+  visible,
+  inputRect,
+}: SlashCommandDropdownProps) {
+  if (!visible || suggestions.length === 0) {
+    return null;
+  }
+
+  const dropdownStyle: React.CSSProperties = inputRect
+    ? {
+        position: 'fixed',
+        top: inputRect.top - 6,
+        left: inputRect.left,
+        width: Math.max(300, inputRect.width),
+        zIndex: 1000,
+        transform: 'translateY(-100%)',
+        maxHeight: 220,
+        overflowY: 'auto',
+      }
+    : {
+        position: 'absolute',
+        bottom: '100%',
+        left: 0,
+        right: 0,
+        zIndex: 1000,
+        maxHeight: 220,
+        overflowY: 'auto',
+      };
```

Comment:
why the ternary operator for styling here? similar question to above, can we do without `inputRect`?